crates/fuel-core/src/graphql_api/indexation/balances.rs

```diff
@@ -0,0 +1,598 @@
+use fuel_core_storage::StorageAsMut;
+use fuel_core_types::{
+    entities::{
+        coins::coin::Coin,
+        Message,
+    },
+    services::executor::Event,
+};
+
+use crate::graphql_api::{
+    ports::worker::OffChainDatabaseTransaction,
+    storage::balances::{
+        CoinBalances,
+        CoinBalancesKey,
+        MessageBalance,
+        MessageBalances,
+    },
+};
+
+use super::error::IndexationError;
+
+fn increase_message_balance<T>(
+    block_st_transaction: &mut T,
+    message: &Message,
+) -> Result<(), IndexationError>
+where
+    T: OffChainDatabaseTransaction,
+{
+    let key = message.recipient();
+    let storage = block_st_transaction.storage::<MessageBalances>();
+    let current_balance = storage.get(key)?.unwrap_or_default().into_owned();
+    let MessageBalance {
+        mut retryable,
+        mut non_retryable,
+    } = current_balance;
+    if message.is_retryable_message() {
+        retryable = retryable.saturating_add(message.amount() as u128);
+    } else {
+        non_retryable = non_retryable.saturating_add(message.amount() as u128);
```

Comment:
Nit: Personally I try to avoid `as` conversions, since they are often fallible and can cause unexpected panics. I believe this one is safe, but I think it both feels and reads better to use `u128::from` instead, to clearly mark that this conversion is infallible

```suggestion
        retryable = retryable.saturating_add(u128::from(message.amount()));
    } else {
        non_retryable = non_retryable.saturating_add(u128::from(message.amount()));
```

Reply:
I'm generally ok with `as` conversions for infallible cases, but `u128::from()` is also good. I updated a couple of places in this commit: 5eb9bb06309ef064a281043fc32a8aaffc916989

Reply:
Thanks :pray: